### ISSUE TITLE
Position is not changed when hovering

### ISSUE THREAD

**Issue by nenadalm** [jbedo/boogiesync-tablet#12]
Hi. It should be probably possible to change mouse coordinates when hovering over the tablet.

> For example, a pen tablet may set BTN_TOOL_PEN to 1 and BTN_TOUCH to 0 while the pen is hovering over but not touching the tablet surface.

(https://www.kernel.org/doc/html/v4.12/input/event-codes.html#ev-key)

I was trying to implement it but once I enabled `BTN_TOOL_PEN`, it stopped working completely.

**Comment by jbedo** [comment]
Is the problem that hovering the pen does not move the mouse, meaning the mouse cursor only shifts when in contact with the tablet? This should be working already, not sure more event codes are needed.

**Comment by nenadalm** [comment]
Hi. yes. that is the problem. Hovering doesn't work for me. Cursor moves only when pen is touching the tablet.

**Comment by jbedo** [comment]
Sorry for delayed response! I'm not sure what to suggest here as I hovering should already correctly work with the code as it currently is. Do you have access to a mac or windows machine? If so you could check if it's a problem specific to your device using the official sync software.

**Comment by nenadalm** [comment]
I don't have access to windows/mac. I tried to print coordinates in the loop though and they changed. Cursor didn't though.

So possibly it's due to version of kernel/libevdev...?

```
$ dnf list installed | grep libevdev
libevdev.x86_64                        1.5.7-1.fc26                @@commandline
libevdev-devel.x86_64                  1.5.7-1.fc26                @fedora

$ uname -r
4.14.14-200.fc26.x86_64

$ lsb_release -a
LSB Version:	:core-4.1-amd64:core-4.1-noarch:cxx-4.1-amd64:cxx-4.1-noarch:desktop-4.1-amd64:desktop-4.1-noarch:languages-4.1-amd64:languages-4.1-noarch:printing-4.1-amd64:printing-4.1-noarch
Distributor ID:	Fedora
Description:	Fedora release 26 (Twenty Six)
Release:	26
Codename:	TwentySix
```

Does it work for you on linux? If so - which distro and version are you using?

**Comment by jbedo** [comment]
Ah interesting. I'm using evdev 1.5.7 and pyevdev 0.7.0 on kernel 4.14.0. How did you set BTN_TOOL_PEN?

**Comment by nenadalm** [comment]
https://github.com/nenadalm/boogiesync-tablet/commit/2bb099ae53c29a1e06b85ab41119fa7d666e422e

coordinates are printed (and changing) no matter if I am hovering or touching the surface of the tablet, but cursor doesn't work with the tool: https://github.com/nenadalm/boogiesync-tablet/commit/2bb099ae53c29a1e06b85ab41119fa7d666e422e#diff-4f655cdde8b656534047109b556824cdR82

Also I am sure it has nothing to do with python versions, because I've tried it in c too (version pushed is without the tool): https://github.com/nenadalm/boogie-board-sync-driver/blob/master/src/usb.c (I wanted to learn more c and get rid of python dependency - as I don't know python and it's a bit easier to compile c than install python and manage it's dependencies...)

**Comment by jbedo** [comment]
You have coordinates being printed when the stylus isn't near the boogieboard? This sounds a bit weird, the read should block when the stylus is not hovering and you shouldn't get any coordinates. What coordinate does it give you when not the pen is absent? The last known position of the pen?

That's probably not the problem though, as evdev should be moving your cursor around anyway. What's the value of `touch` when hovering and when in contact?

**Comment by nenadalm** [comment]
> You have coordinates being printed when the stylus isn't near the boogieboard? 

No, I have coordinates printed when stylus is touching or is almost touching the board (if the pen is too far away - no coordinates are printed). Coordinates printed are coordinates of the pen (in place it is touching the board or in place it is almost touching the board).

> What's the value of touch when hovering and when in contact?

Touch is `1` when in contact and `0` when hovering.

**Comment by nenadalm** [comment]
I just tried on Ubuntu 16.04 and it works as expected (I might investigate more in a future). Should I close this ticket?

**Comment by jbedo** [comment]
Yeah did sound like an evdev problem, thanks for debugging :).

**Comment by nenadalm** [comment]
Hi. So I thought I would take a look into this again and now it works also on Fedora.

I found that I can list inputs with
```shell
$ sudo libinput list-devices
```

in which output I saw some errors like `missing BTN_STYLUS`, something about invalid resolution...

So after fixing these it works!!! I was playing with it in c as I don't feel like installing python dependencies, but should be straiforward to move into python. From changes I remember:
- set resolution in absolute positioning from `0` to `1` (https://github.com/nenadalm/boogie-board-sync-driver/blob/23c910d3593fab93895bed73b5b77cc6d9ea98e3/src/usb.c#L95)
- use BTN_STYLUS instead of BTN_STYLUS2 (https://github.com/nenadalm/boogie-board-sync-driver/blob/23c910d3593fab93895bed73b5b77cc6d9ea98e3/src/usb.c#L160)
- set INPUT_PROP_POINTER property (https://github.com/nenadalm/boogie-board-sync-driver/blob/23c910d3593fab93895bed73b5b77cc6d9ea98e3/src/usb.c#L118)

just in case you're interested.
